### PR TITLE
Accumulated diagnostics

### PR DESCRIPTION
**Pull request by anotherjohnwong**
Adding fixed buckets for accumulated total precipitations, snow, and ice. This diagnostics module will potentially serve as a good entry point for adding buckets for non-precipitation quantities, e.g. latent heat and radiations, which are also needed for post-processing.

### CODE REVIEW

File: src/core_atmosphere/diagnostics/accumulated_diagnostics.F [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+            ! Precipitation accumulation
+            precip = max(0., rainncv(iCell) + raincv(iCell))
+            precip15m(iCell) = precip15m(iCell) + precip
+
+            ! Snow accumulation
+            snow = snowncv(iCell) + graupelncv(iCell)
+            snow15m(iCell) = snow15m(iCell) + snow * snow_ratio(iCell)
+
+            ! Ice/freezing rain accumulation
+            if (t2m(iCell) < 273.16) then
```
Comment (bawilt0920) [inline src/core_atmosphere/diagnostics/accumulated_diagnostics.F]: testing?

Comment (anotherjohnwong) [reply src/core_atmosphere/diagnostics/accumulated_diagnostics.F]: Yes 👍 

OK, so this scaling factor is copied from gridcalc's calculation_functions.C iceAlg. I asked @tahutchinson about it and he said it is a scaling factor through trial and error to account for the freezing potential? I probably said it all wrong though.

Comment (bawilt0920) [reply src/core_atmosphere/diagnostics/accumulated_diagnostics.F]: I am not sure why the amount would be reduced, but maybe the issue here is sleet vs freezing rain?

Comment (anotherjohnwong) [reply src/core_atmosphere/diagnostics/accumulated_diagnostics.F]: @tahutchinson For documentation purposes, can you summarize what you told me the other day about the reduction below/after 27F in a few words? I'll add that to the code as inline comment (or you can).

Comment (tahutchinson) [reply src/core_atmosphere/diagnostics/accumulated_diagnostics.F]: Yes, the amount of aggregated ice is set at 0.8 * liquid when temp < 27F, and 0.66*liquid when 27 < temp < 32.  The idea behind this is that the amount of liquid that falls and accumulates frozen is not the total amount of rainfall, since some of the rainfall will just flow away.  In other words, if the temp is 31 degrees, and you get an inch of rain, you won't get a full inch of ice on tree branches, windshields, etc.

Comment (tahutchinson) [reply src/core_atmosphere/diagnostics/accumulated_diagnostics.F]: I've also added the comment to code and pushed it.
